### PR TITLE
Workaround for volume shadow copy in WSL1

### LINKED ISSUE #6024: EROFS due to O_NOATIME with volume shadow copy in WSL1
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Bug

#### Your borg version (borg -V).

borg 1.1.17

#### Operating system (distribution) and version.

Windows 10 21h1

#### Hardware / network configuration, and filesystems used.

WSL1 drvfs

#### Can you reproduce the problem? If so, describe how. If not, describe troubleshooting steps you took before opening the issue.

A volume shadow copy is made available in WSL1 with these steps:

1. Create a symlink in Windows with the shadow copy's UNC path.
    `mklink /d c \\?\GLOBALROOT\Device\HarddiskVolumeShadowCopy1\`
2. Map symlink to a drive letter.
    `subst z: c`
3. Mount the drive in WSL1.
    `mount -t drvfs -o ro z: c`

When borg encounters a file in this mounted volume owned by the invoking user, the file is opened with `O_NOATIME` among other flags, which produces this error `open: [Errno 30] Read-only file system`. When the file is not owned by the invoking user, `O_NOATIME` is not used and it works fine.

Can be confirmed with this patch

```diff
diff --git a/src/borg/archive.py b/src/borg/archive.py
index 93385c9d..4c08f9a7 100644
--- a/src/borg/archive.py
+++ b/src/borg/archive.py
@@ -47,7 +47,7 @@
 has_link = hasattr(os, 'link')

 flags_normal = os.O_RDONLY | getattr(os, 'O_BINARY', 0)
-flags_noatime = flags_normal | getattr(os, 'O_NOATIME', 0)
+flags_noatime = flags_normal


 class Statistics:
```

### PR DESCRIPTION
**Pull request by cuevavirus**
Fixes #6024 for master. Similar to patch #6027 for maint-1.1 submitted previously.

### CODE REVIEW

File: src/borg/helpers/fs.py [hunk 1]
```diff
@@ -286,6 +286,13 @@ def os_open(*, flags, path=None, parent_fd=None, name=None, noatime=False):
                 raise
             # Was this EPERM due to the O_NOATIME flag? Try again without it:
             fd = os.open(fname, _flags_normal, dir_fd=parent_fd)
+        except OSError as exc:
+            # O_NOATIME causes EROFS when accessing a volume shadow copy in WSL1
+            from . import workarounds
```
Comment (ThomasWaldmann) [inline src/borg/helpers/fs.py:291]: did you try to do this import on global level instead of here?

Comment (cuevavirus) [reply src/borg/helpers/fs.py:291]: Yes in that case import fails due to circular import.

https://github.com/borgbackup/borg/blob/bf392367f159fd81494a9d76bb646e7f3ab37132/src/borg/helpers/__init__.py#L12

Comment (ThomasWaldmann) [reply src/borg/helpers/fs.py:291]: ok, i had a similar effect recently, too.

**Comment by ThomasWaldmann** [comment]
Thanks!

**Pull request merged**